src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -873,26 +920,20 @@
   [{:keys [projections]}]
   (->> projections
        (remove (comp :query-only? val))
-       keys
-       sort))
-
-(defn extract-fields
-  "Returns all fields from a projection.  Returns nil for fields which
-  are query-only? since these can't be projected either."
-  [[name {:keys [field]}] entity]
-  [field name])
+       keys))
```

Comment:
There's a (probably not elaborate enough) comment in the commit message for this change, but it's because I believe the engine relies on the correlation between the ordering of the projections and the columns, and so sorting was at best extra work, and at worst, might break the correlation unless both sides always sort exactly the same.  I think I had an example that would break, but now can't remember it.  (Or maybe I've just misunderstood the situation...)